src/App.js

```diff
@@ -0,0 +1,108 @@
+import './App.css';
+import {Checklist} from './Checklist.js'
+import {AddTaskPage} from "./AddTaskPage";
+import {useEffect, useState} from "react";
+import {IncompleteTasksOnly} from "./IncompleteTasksOnly";
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+import {ModifyTaskPage} from "./ModifyTaskPage";
+
+export function App(props) {
+
+    let [data, setData] = useState(props.initialData);
+    let [itemsToShow, setItemsToShow] = useState("both");
+    const initialCompletedItems = data.map((item) => item.completed ? item.id : undefined);
+    let [completedItems, setCompletedItems] = useState(initialCompletedItems);
+    let [componentsToRender, setComponentsToRender] = useState(<Checklist items={data} handleChangeField={handleChangeField} modifyTask={modifyTask} completedItems={completedItems} changeCompletedItems={setCompletedItems}/>);
```

Comment:
I don't understand what componentsToRender is for.

Reply:
It updates the components to render based on clicking buttons and adding new data. For example, when you click "hide completed items," it changes componentsToRender from the initial checklist to the checklist without completed items.